hypershift-operator/controllers/nodepool/manifests.go

```diff
@@ -37,6 +40,58 @@ func machineHealthCheck(nodePool *hyperv1.NodePool, controlPlaneNamespace string
 	}
 }
 
+func KubevirtMachineTemplate(infraName string, hostedCluster *hyperv1.HostedCluster, nodePool *hyperv1.NodePool, controlPlaneNamespace string) (*capikubevirt.KubevirtMachineTemplate, string) {
+	guestQuantity := apiresource.MustParse("4Gi")
+	kubevirtMachineTemplate := &capikubevirt.KubevirtMachineTemplate{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: map[string]string{
+				nodePoolAnnotation: ctrlclient.ObjectKeyFromObject(nodePool).String(),
+			},
+			Namespace: controlPlaneNamespace,
+		},
+		Spec: capikubevirt.KubevirtMachineTemplateSpec{
+			Template: capikubevirt.KubevirtMachineTemplateResource{
+				Spec: capikubevirt.KubevirtMachineSpec{
+					VMSpec: kubevirtv1.VirtualMachineInstanceSpec{
+						Domain: kubevirtv1.DomainSpec{
+							CPU:    &kubevirtv1.CPU{Cores: 2},
+							Memory: &kubevirtv1.Memory{Guest: &guestQuantity},
+							Devices: kubevirtv1.Devices{
```

Comment:
What's the plan here for configuration? I'd imagine we need to expand the `NodePoolPlatform` struct in `api/v1alpha1/nodepool_types.go` with a KubeVirt platform type of some sort. 

Is that something we should tackle in this PR? Maybe it could be as simple as exposing resource request/limits field so cpu+mem can be configured. 

```
// NodePoolPlatform specifies the underlying infrastructure provider for the
// NodePool and is used to configure platform specific behavior.
type NodePoolPlatform struct {   
        // Type specifies the platform name.
        //              
        // +unionDiscriminator
        // +immutable
        Type PlatformType `json:"type"`

        // AWS specifies the configuration used when operating on AWS.
        //      
        // +optional
        // +immutable
        AWS *AWSNodePoolPlatform `json:"aws,omitempty"`

        KubeVirt *KubeVirtNodePoolPlatform `json:"aws,omitempty"`
}

type KubeVirtNodePoolPlatform struct {
        // Requests is a description of the initial vmi resources.
        // Valid resource keys are "memory" and "cpu".
        // +optional
        Requests v1.ResourceList `json:"requests,omitempty"`
}

```

Reply:
Changed it - added VMITemplate to the KubevirtPlatformNodePool